lib/erb_lint/linter_config.rb

```diff
@@ -1,31 +1,58 @@
 # frozen_string_literal: true
 
 require 'active_support'
+require 'smart_properties'
 
 module ERBLint
   class LinterConfig
-    def initialize(config = {})
-      @config = config.dup.deep_stringify_keys
-    end
+    include SmartProperties
 
-    def to_hash
-      @config.dup
+    class Error < StandardError; end
+
+    class << self
+      def array_of?(klass)
+        lambda { |value| value.is_a?(Array) && value.all? { |s| s.is_a?(klass) } }
+      end
+
+      def to_array_of(klass)
+        lambda { |entries| entries.map { |entry| klass.new(entry) } }
+      end
     end
 
-    def [](key)
-      @config[key.to_s]
+    property :enabled, accepts: [true, false], default: false, reader: :enabled?
```

Comment:
Do we really need a gem to check two arguments?

Reply:
each of the linters have their own configuration properties which inherit from `LinterConfig`, it is more than 2 arguments...

Reply:
oh, and we already have better-html depending on it so it is fine.